crd/apis/network/v1alpha1/gkenetworkparamset_types.go

```diff
@@ -64,14 +64,77 @@ type NetworkRanges struct {
+
+const (
+	// SubnetNotFound indicates that the specified subnet was not found.
+	SubnetNotFound GKENetworkParamSetReason = "SubnetNotFound"
+	// SecondaryRangeNotFound indicates that the specified secondary range was not found.
+	SecondaryRangeNotFound GKENetworkParamSetReason = "SecondaryRangeNotFound"
+	// DeviceModeCantBeUsedWithSecondaryRange indicates that device mode was used with a secondary range.
+	DeviceModeCantBeUsedWithSecondaryRange GKENetworkParamSetReason = "DeviceModeCantBeUsedWithSecondaryRange"
+	// DeviceModeVPCAlreadyInUse indicates that the VPC is already in use by another GKENetworkParamSet resource.
+	DeviceModeVPCAlreadyInUse GKENetworkParamSetReason = "DeviceModeVPCAlreadyInUse"
+	// DeviceModeCantUseDefaultVPC indicates that a device mode GKENetworkParamSet cannot use the default VPC.
+	DeviceModeCantUseDefaultVPC GKENetworkParamSetReason = "DeviceModeCantUseDefaultVPC"
+	// DPDKUnsupported indicates that DPDK device mode is not supported on the current cluster.
+	DPDKUnsupported GKENetworkParamSetReason = "DPDKUnsupported"
+)
+
+// GNPNetworkParamsReadyReason defines the set of reasons that explains
+// the ParamsReady condition on the referencing Network resource.
+type GNPNetworkParamsReadyReason string
+
+const (
+	// L3SecondaryMissing indicates that the L3 type Network resource is
+	// referencing a GKENetworkParamSet with secondary range unspecified.
+	L3SecondaryMissing GNPNetworkParamsReadyReason = "L3SecondaryMissing"
+	// L3DeviceModeExists indicates that the L3 type Network resource is
+	// referencing a GKENetworkParamSet with device mode specified.
+	L3DeviceModeExists GNPNetworkParamsReadyReason = "L3DeviceModeExists"
+	// DeviceModeMissing indicates that the Device type Network resource is
+	// referencing a GKENetworkParamSet with device mode unspecified.
+	DeviceModeMissing GNPNetworkParamsReadyReason = "DeviceModeMissing"
+	// DeviceSecondaryExists indicates that the Device type Network resource is
+	// referencing a GKENetworkParamSet with a secondary range specified.
+	DeviceSecondaryExists GNPNetworkParamsReadyReason = "DeviceSecondaryExists"
+)
+
 // GKENetworkParamSetStatus contains the status information related to the network.
 type GKENetworkParamSetStatus struct {
 	// PodCIDRs specifies the CIDRs from which IPs will be used for Pod interfaces
 	// +optional
 	PodCIDRs *NetworkRanges `json:"podCIDRs,omitempty"`
 
 	// Conditions is a field representing the current conditions of the GKENetworkParamSet.
-	Conditions []metav1.Condition `json:"conditions"`
+	//
+	// Known condition types are:
+	//
+	// * "Ready"
+	//
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+
+	// NetworkName specifies which Network object is currently referencing this GKENetworkParamSet
```

Comment:
how is this reference obtained?

Reply:
This change is a part of #507. The reference is obtained from GNP controller in a later PR.